File: 1984-minimum-difference-between-highest-and-lowest-of-k-scores/1984-minimum-difference-between-highest-and-lowest-of-k-scores.py
class Solution(object):
    def minimumDifference(self, nums, k):
        ans = float("inf")
        nums.sort()
        l=0
        for r in range(len(nums)):
            if(r-l == k):
                l+=1
            if(r-l+1 == k):
                ans=min(ans, nums [r]-nums[l])
        return ans

File: 1984-minimum-difference-between-highest-and-lowest-of-k-scores/1984-minimum-difference-between-highest-and-lowest-of-k-scores.java
class Solution {
    public int minimumDifference(int[] nums, int k) {
        int n = nums.length;
        Arrays.sort(nums);
        int ans = Integer.MAX_VALUE;
        for (int i = 0; i <= n - k; i++) {
            int j = i + k - 1;
            int temp = nums[j] - nums[i];
            ans = Math.min(ans, temp);
        }
        return ans;
    }
}